src/index/diskann/diskann.cc

```diff
@@ -502,18 +503,19 @@ expected<DataSetPtr>
 DiskANNIndexNode<T>::Search(const DataSet& dataset, const Config& cfg, const BitsetView& bitset) const {
     if (!is_prepared_.load() || !pq_flash_index_) {
         LOG_KNOWHERE_ERROR_ << "Failed to load diskann.";
-        return Status::empty_index;
+        return expected<DataSetPtr>::Err(Status::empty_index, "DiskANN not loaded");
     }
 
     auto search_conf = static_cast<const DiskANNConfig&>(cfg);
     if (!CheckMetric(search_conf.metric_type.value())) {
-        return Status::invalid_metric_type;
+        return expected<DataSetPtr>::Err(Status::invalid_metric_type, "unsupported metric type");
     }
     auto max_search_list_size = std::max(kSearchListSizeMaxValue, search_conf.k.value() * 10);
     if (search_conf.search_list_size.value() > max_search_list_size ||
         search_conf.search_list_size.value() < search_conf.k.value()) {
         LOG_KNOWHERE_ERROR_ << "search_list_size should be in range: [topk, max(200, topk * 10)]";
-        return Status::out_of_range_in_json;
+        return expected<DataSetPtr>::Err(Status::out_of_range_in_json,
+                                         "search_list_size should be in range: [topk, max(200, topk * 10)]");
```

Comment:
The input value and the specific value of topk should also be included in the error message.